lntest/itest/lnd_routing_test.go

```diff
@@ -1216,14 +1165,73 @@ func testInvoiceRoutingHints(net *lntest.NetworkHarness, t *harnessTest) {
 		}
 	}
 
-	if aliceBobChanID == 0 {
-		t.Fatalf("channel between alice and bob not found")
+	require.NotZero(t.t, aliceBobChanID,
+		"channel between alice and bob not found")
+
+	checkInvoiceHints := func(invoice *lnrpc.Invoice) {
+		// Due to the way the channels were set up above, the channel
+		// between Alice and Bob should be the only channel used as a
+		// routing hint.
+		var predErr error
+		var decoded *lnrpc.PayReq
+		err := wait.Predicate(func() bool {
+			ctxt, _ := context.WithTimeout(ctxb, defaultTimeout)
+			resp, err := net.Alice.AddInvoice(ctxt, invoice)
+			if err != nil {
+				predErr = fmt.Errorf(
```

Comment:
nit: we usually wrap errors, logs, in a compact mode, sth like,
```go
fmt.Errorf("unable to add "+
"invoice: %w", err)
```
This is just FYI for your future reference tho, as the itest will soon be changed so don't bother changing it here now.

Reply:
Good to know, thanks!